node/internal/util.js

```diff
@@ -92,10 +93,56 @@ function createDeferredPromise() {
   return { promise, resolve, reject };
 }
 
+// Keep a list of deprecation codes that have been warned on so we only warn on
+// each one once.
+const codesWarned = new Set();
+
+// Mark that a method should not be used.
+// Returns a modified function which warns once by default.
+// If --no-deprecation is set, then it is a no-op.
+export function deprecate(fn, msg, code) {
+  if (process.noDeprecation === true) {
```

Comment:
It seems we need to import `process`

Reply:
This will most likely cause circular dependency; I suggest to move `doEmitWarning` and `emitWarning` functions from `process.ts` to `_process/process.ts` and import these function from that module, instead of importing `process.ts` directly.

Reply:
`error: Uncaught ReferenceError: Cannot access 'ERR_IPC_CHANNEL_CLOSED' before initialization` does not cause after original test and `process` are deleted.https://github.com/denoland/deno_std/pull/1697/commits/fdf67e459415a7e81839ce9f32fd92d64704d8d3

`process` doesn't need to import for `parallel/test-util-deprecate.js` to pass. https://github.com/denoland/deno_std/pull/1697/commits/8635408a37706c6323f2c1d5234aa75ef4c784d2